app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdConfig.kt

```diff
@@ -0,0 +1,14 @@
+package org.simple.clinic.scanid
+
+import org.simple.clinic.remoteconfig.ConfigReader
+
+data class ScanSimpleIdConfig(val useNewQrScanner: Boolean) {
+
+  companion object {
+
+    fun read(configReader: ConfigReader): ScanSimpleIdConfig {
+      val useNewQrScanner = configReader.boolean("use_new_qr_scanner", true)
+      return ScanSimpleIdConfig(useNewQrScanner = useNewQrScanner)
+    }
+  }
```

Comment:
I've been thinking of getting rid of companion functions for constructing config instances. We were using them earlier because we were injecting `Observable` instances, but now that we inject them directly, I think we can replace this with a secondary constructor.

```kotlin
@Inject constructor(configReader: ConfigReader): this(
      useNewQrScanner = configReader.boolean("use_new_qr_scanner", true)
)
```

With this, we can tell Dagger to use this constructor for injection and avoid having to construct a new module completely.

What do you think?